### ISSUE TITLE
Missing face/textures with high amount of vertices in a csv object

### ISSUE THREAD

**Issue by Railfan007** [leezer3/OpenBVE#448]
### Description
My issue is faces and the textures are not being displayed. If I move the code to the top of the file, it is displayed, however if its moved to the bottom, they wont be displayed. It seems like the objectviewer/Routeviewer/OpenBVE hits a limit. I'm guessing its either a Vertex count cap of 9724, or it is hitting a limit in terms of the amount of lines of code it will display. Also, when I place this same object into a route and load it in Routeviewer/OpenBVE with a train, it shows up the same way.

### Reproduction
Here is a link to the object and its corresponding textures: https://www.dropbox.com/s/obiuyiffb35i4t0/Hits%20BVE%20limit.zip

Here is an image with the section of code at the top of the file: https://cdn.discordapp.com/attachments/451037457475960860/695426783637667905/unknown.png

Here is an image with the same lines of code moved to the bottom of the file: https://cdn.discordapp.com/attachments/451037457475960860/695426836146159686/unknown.png

### Related information
* Operating system: Win 10
* Method of control: Keyboard and mouse

**Comment by leezer3** [comment]
I don't think it's either actually, although I can't quite put my finger on what's going on.

If I add TranslateAll 0,5,0 followed by another complete copy of your code (or anything else for that matter), the end of your rail breaks, but everything else is OK.
This tells me that the parser is working alright, so it's something in the structure of your object somewhere.
![twins](https://user-images.githubusercontent.com/7688245/78943650-2f085a80-7ab4-11ea-8fc2-82480e44658c.png)

My best guess without digging through the entire thing would be that you've either got a missing MeshBuilder somewhere in the file (In this case, the texture commands are essentially ignored, and the vertex numbering carries on), or something funny happening with the vertex counts somewhere.

TLDR:
Check the preceeding code, *not* the broken code when it's at the bottom of the file.

**Comment by leezer3** [comment]
Did a quick little bit of digging.

Your file is missing the *first* CreateMeshBuilder statement, hence the glitch. When you copy the lot to the bottom of the file, these verticies are considered to be part of the preceeding MeshBuilder. 

Next build will warn that you've declared vertices outside the initial MeshBuilder.
Detecting variations on this is likely to be a pain in the neck- The following is completely valid code from POV of the parser:

```
[MeshBuilder]
Vertex 0,0,0
Vertex 0,1,0
Vertex 0,1,1
Vertex 0,0,1
Face 0,1,2,3
Vertex 1,0,1
Vertex 1,1,1
Face 5,4,3,2
```

Whether anyone has actually structured something like this I've got no idea, and I don't especially want to go down that particular rabbit hole unless I have to......

**Comment by Railfan007** [comment]
> Your file is missing the first CreateMeshBuilder statement, hence the glitch. When you copy the lot to the bottom of the file, these verticies are considered to be part of the preceeding MeshBuilder.

Ahhh.

> Next build will warn that you've declared vertices outside the initial MeshBuilder.
Detecting variations on this is likely to be a pain in the neck- The following is completely valid code from POV of the parser .... Whether anyone has actually structured something like this I've got no idea, and I don't especially want to go down that particular rabbit hole unless I have to......

This is exactly why I opened the issue, as I was confused to why it processed fine. I doubt anyone has structured objects like this, and it was just an error on my part. My apologies.

Thanks again for helping!!
